examples/quickstart/executor_with_thread_hooks.cpp

```diff
@@ -0,0 +1,228 @@
+//  Copyright (c) 2020 Hartmut Kaiser
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+///////////////////////////////////////////////////////////////////////////////
+// The purpose of this example is to show how to rite an executor that wraps
+// any other executor and adds a hook into thread start and thread exit allowing
+// to associate custom thread data with the tasks that are created by the
+// underlying executor.
+
+#include <hpx/hpx_main.hpp>
+
+#include <hpx/algorithms.hpp>
+#include <hpx/execution.hpp>
+#include <hpx/functional.hpp>
```

Comment:
Just inspect left on this one: https://173022-4455628-gh.circle-artifacts.com/0/hpx_inspect_report.html. It needs `<cstddef>` and `hpx/assertion.hpp`.